features/steps/print_steps.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime, timedelta
+from pathlib import Path
+from time import sleep
+
+from behave import step
+
+from config import Config
+from utilties.decrypt import decrypt_message
+from utilties.mappings import PACK_CODE_TO_SFTP_DIRECTORY, PACK_CODE_TO_ACTION_TYPE
+from utilties.sftp_utility import SftpUtility
+
+
+# timeout in concourse to stop it going on forever
+@step('all the initial contact print files are produced on the SFTP')
+def wait_for_print_files(context):
+    with SftpUtility() as sftp:
+        while True:
+            context.all_initial_print_sftp_paths = fetch_all_print_files_paths(sftp, context)
+            if context.all_initial_print_sftp_paths:
+                context.produced_print_file_time = datetime.utcnow()
+                context.print_file_production_run_time = context.produced_print_file_time \
+                    - context.action_rule_trigger_time
+                # TODO nicer way to print within behave
+                print(f'Time from action rule trigger to all print files produced: '
+                      f'[{str(context.print_file_production_run_time)}]\n')
+                break
+        sleep(int(Config.SFTP_POLLING_DELAY))
+
+
+def fetch_all_print_files_paths(sftp, context):
+    print_file_paths = []
+    print_file_paths.extend([Path(Config.SFTP_QM_DIRECTORY).joinpath(str(file_name.filename)) for file_name in
+                             sftp.get_all_print_files(Config.SFTP_QM_DIRECTORY, context.test_start_local_datetime)])
+    print_file_paths.extend([Path(Config.SFTP_PPO_DIRECTORY).joinpath(str(file_name.filename)) for file_name in
+                             sftp.get_all_print_files(Config.SFTP_PPO_DIRECTORY, context.test_start_local_datetime)])
+
+    for pack_code in PACK_CODE_TO_SFTP_DIRECTORY.keys():
+        matching_csv_files = [print_file_path for print_file_path in print_file_paths
+                              if print_file_path.name.startswith(pack_code)
+                              and print_file_path.name.endswith('.csv.gpg')]
+        matching_manifest_files = [print_file_path for print_file_path in print_file_paths
+                                   if print_file_path.name.startswith(pack_code)
+                                   and print_file_path.name.endswith('.manifest')]
+        if len(matching_csv_files) != 1 and len(matching_manifest_files) != 1:
```

Comment:
Just a quick question - Is this safe to do with a randomly generated sample file if we can't guarantee treatment codes? E.g. I loaded [this sample file](https://raw.githubusercontent.com/ONSdigital/census-rm-acceptance-tests/7fe481171500d4f3adce745b10a4ab6464a4dbce/resources/sample_files/sample_input_census_spec_england_questionnaire.csv) and as there is no pack code `P_IC_ICL1` output, the test continuously loops.

Reply:
So using the `generate_sample.py` on sample loader creates a sample file using all the treatment codes. With the acceptance test one, its only using the english questionnaire file so it's just looping and waiting for all the print files to be generated which they never will with that sample file. I think any sample that we use with this will have all the treatment codes on it and not a specific set.